src/layout/layout.rs

```diff
@@ -420,6 +420,73 @@ impl Layout {
         self
     }
 
+    /// Split the rect into a number of sub-rects according to the given [`Layout`]`.
+    ///
+    /// An ergonomic wrapper around [`Layout::split`] that returns an array of `Rect`s instead of
+    /// `Rc<[Rect]>`.
+    ///
+    /// This method requires the number of constraints to be known at compile time. If you don't
+    /// know the number of constraints at compile time, use [`Layout::split`] instead.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the number of constraints is not equal to the length of the returned array.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::prelude::*;
+    /// # fn render(frame: &mut Frame) {
+    /// let area = frame.size();
+    /// let layout = Layout::vertical([Constraint::Length(1), Constraint::Min(0)]);
+    /// let [top, main] = layout.areas(area);
+    ///
+    /// // or explicitly specify the number of constraints:
+    /// let areas = layout.areas::<2>(area);
+    /// # }
+    pub fn areas<const N: usize>(&self, area: Rect) -> [Rect; N] {
+        self.split(area)
+            .to_vec()
+            .try_into()
+            .expect("invalid number of rects")
```

Comment:
Maybe like this for consistency with `.spacers()`?

```suggestion
        let (areas, _) = self.split_with_spacers(area);
        areas
            .to_vec()
            .try_into()
            .expect("invalid number of rects")
```

Reply:
makes sense